src/Language/PureScript/AST/Declarations.hs

```diff
@@ -451,7 +473,7 @@ data Declaration
   -- |
   -- A value declaration (name, top-level binders, optional guard, value)
   --
-  | ValueDeclaration SourceAnn Ident NameKind [Binder] [GuardedExpr]
+  | ValueDeclaration {-# UNPACK #-} !(ValueDeclarationData [GuardedExpr])
```

Comment:
I would avoid adding UNPACK pragmas and bang annotations until profiling shows it to be necessary. It's not always an optimization (although, I'll admit, it is usually in these sorts of cases).